dspace-uclouvain/src/main/java/org/dspace/uclouvain/services/UCLouvainAffiliationEntityRestServiceImpl.java

```diff
@@ -105,67 +72,92 @@ private void createAffiliationsStructure(List<AffiliationEntityRestModel> models
+            if (itemID != null) {
+                model.UUID = UUID.fromString(itemID);
+                model.name = documentExtract(document, "title");
+                model.acronym = documentExtract(document, "oairecerif.acronym");
+                model.type = documentExtract(document, "dc.type");
+
+                String selectable = documentExtract(document, "organization.isSelectable");
+                model.isSelectable = (selectable != null) ? selectable.equals("true"): true;
+
+                // Extract parent UUID
+                String parentAuthority = documentExtract(document, "organization.parentOrganization_authority");
+                try {
+                    if (parentAuthority != null) {
+                        model.parent = UUID.fromString(parentAuthority);
+                    }
+                } catch (Exception e) {
+                    // LOG error, authority should be a valid UUID string
+                    this.logger.error("Error while parsing parent UUID of OrgUnit with ID: " + itemID + " - error: " + e);
                 }
-            } catch (Exception e) {
-                // LOG error, authority should be a valid UUID string
-                this.logger.error("Error while parsing parent UUID of OrgUnit with ID: " + item.getID() + " - error: " + e);
+                model.children = new ArrayList<AffiliationEntityRestModel>();
+                models.add(model);
+            } else {
+                // Log as a warning that an item has no UUID and add a list of all his metadata for investigation purpose.
+                Map<String, Collection<Object>> valuesMap =  document.getFieldValuesMap();
+                String documentContent = "";
+                for (String key: valuesMap.keySet()){
+                    String keyValues = key + "[" + valuesMap.get(key).stream().map(value -> value.toString()).collect(Collectors.joining(", ")) + "]";
+                    documentContent = documentContent + keyValues;
+                }
+                this.logger.warn("Could not find UUID for SolrDocument with following metadata:\n" + documentContent);
             }
-            model.children = new ArrayList<AffiliationEntityRestModel>();
-            models.add(model);
         }
         return models;
     }
 
+    /**
+     * Little util method to extract the first value of a SOLR document field.
+     * If nothing is found for a field, return null. Else return the value as a string.
+     * 
+     * @param document: The SOLR document ot search values in.
+     * @param field: The field to search for.
+     * @return: The first value in the document for the given field or null if not found.
+     */
+    private String documentExtract(SolrDocument document, String field) {
+        Object value = document.getFirstValue(field);
+        if (value != null) {
+            return value.toString();
+        }
+        return null;
+    }
```

Comment:
It could be great to have a default return value and use a ternary for return.
```java
private String documentExtract(SolrDocument document, String field) {
    return this.documentExtract(document, field, null);
}
private String documentExtract(SolrDocument document, String field, String defaultValue) {
    Object value = document.getFirstValue(field);
    return (value != null)
       ? value.toString()
       : defaultValue;
}
```